tests/unit/python/foglamp/services/core/scheduler/test_scheduler.py

```diff
@@ -0,0 +1,841 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+import asyncio
+import datetime
+import os
+import time
+import uuid
+import pytest
+import json
+from unittest import mock
+from unittest.mock import patch
+from aiohttp.test_utils import make_mocked_request
+from aiohttp.streams import StreamReader
+from multidict import CIMultiDict
+
+from foglamp.services.core.scheduler.scheduler import Scheduler, _FOGLAMP_ROOT
+from foglamp.services.core.scheduler.entities import IntervalSchedule, Task, Schedule, TimedSchedule, ManualSchedule, \
+    StartUpSchedule
+from foglamp.services.core.scheduler.exceptions import ScheduleNotFoundError
+# from tests.unit.python.foglamp.services.common.storage_layer_fixture import Storage as Storage
+
+__author__ = "Amarendra K Sinha"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+@pytest.allure.feature("unit")
+@pytest.allure.story("scheduler")
+class TestScheduler:
+    @classmethod
+    def setup_class(cls):
+        cls.Scheduler = Scheduler(core_management_host=None, core_management_port=None)
+        cls.Scheduler._storage = Storage()
+
+    @classmethod
+    def teardown_class(cls):
+        pass
+
+    @pytest.mark.asyncio
+    async def test__resume_check_schedules(self):
+        # Check IF part
+        with patch.object(TestScheduler.Scheduler, '_scheduler_loop_sleep_task', asyncio.Task(asyncio.sleep(5))):
```

Comment:
In this test, we are leveraging a single Scheduler object created in setup_class. Rather, the Scheduler object should be created within the test code (or at least per test). In general, this seems like a very long unit test. It may be that the original code needs to be refactored to be more testable.